src/jit/compiler.h

```diff
@@ -604,13 +638,19 @@ class  LclVarDsc
     bool                lvIsMultiregStruct()
     {
 #if FEATURE_MULTIREG_ARGS_OR_RET
-#ifdef _TARGET_ARM64_
-        if ((TypeGet() == TYP_STRUCT) &&
-            (lvSize()  == 2 * TARGET_POINTER_SIZE))
+        if (TypeGet() == TYP_STRUCT)
         {
-            return true;
+            if (lvIsHfa() && (lvHfaSlots() > 1))
+            {
+                return true;
+            }
+#if defined(_TARGET_ARM64_)
+            else if (lvSize() == (2 * TARGET_POINTER_SIZE))
```

Comment:
This definition seems to be not consistent with the change made to argOrReturnTypeForStruct(). That is structs of size 9-16 bytes are passed by value in two registers on Arm64?  Does this condition need modification?